### DISCUSSION TITLE
Why Does The Mac Version Of C Debugging Have printf Delay While On Windows It Works Fine

### DISCUSSION THREAD

**Discussion by andychyou** [microsoft/vscode-cpptools#10369] [Q&A]
I have been facing the problem of printf() function delaying to show the outputs on the console. For example, when I have the program on a breakpoint in the printf(“hello world”) line and step over, the “hello world” doesn’t show up right after the line is executed. It is very weird as although the printf is executed, the output doesn’t show up on console right away.

It seems that I am not the only one witnessing this problem. Down below is the link to another user demonstrating what the problem looks like. He thinks the delay is due to the cache waiting. 
[https://www.youtube.com/watch?v=wKjFVyDbSpA&t=9m42s](https://www.youtube.com/watch?v=wKjFVyDbSpA&t=9m42s) 

But the funny thing is, if I have an ‘\n’ at the end of the string, it works fine. Meaning, executing printf(“hello world”) outputs nothing while printf(“hello world\n”) does print right away.

I don’t understand why this only happens on my Mac. On my WIndows machine, the printf will always return the output on the console right after execution. My current only guess on why the difference is because on Windows I use mingw64 and on Mac clang lldb for debugging. But it is so weird that if it is the nature of lldb with the printf cache delay why does it only happen on VS Code and not on XCode? On XCode it works fine.

I would appreciate if anyone could explain to me on why there is an output delay and if possible, how to fix it.

Here are some screenshots for reference.

<img width="545" alt="스크린샷 2023-01-10 오후 10 01 25" src="https://user-images.githubusercontent.com/48653304/211560624-120feda5-d252-4fe0-9051-0c365d8d44ad.png">
Mac with '\n'
<br>

<img width="437" alt="스크린샷 2023-01-10 오후 10 01 51" src="https://user-images.githubusercontent.com/48653304/211560679-6165d156-a6e8-479c-ad7a-00b025b43edc.png">
Mac without '\n'
<br>

![1673356120309](https://user-images.githubusercontent.com/48653304/211560714-d77fdfff-49d0-43d8-9b60-3c5ae2e76cca.png)
Windows working fine

**Comment by sean-mcmanus** [comment]
@WardenGnaw Do you know about this?

**Comment by mixaanttila31** [comment]
The situation is the same when using gcc from shell:
Apple clang version 15.0.0 (clang-1500.1.0.2.5)
Target: arm64-apple-darwin23.3.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

![image](https://github.com/microsoft/vscode-cpptools/assets/148260098/b90c0840-10ca-4f33-a34b-cf3ecee6cd07)

yields
hello world!!
< 10s delay >
0123456789
